File: src/main/java/tweetzor/core/Location.java
package tweetzor.core;

import twitter4j.GeoLocation;

public class Location {
  
  // Fields
  // ------

  private double latitude;
  private double longitude;

  // Constructors
  // ------------

  public Location(double latitude, double longitude) {
    this.latitude = latitude;
    this.longitude = longitude;
  }
  
  public Location(GeoLocation loc) {
    this.latitude = loc.getLatitude();
    this.longitude = loc.getLongitude();
  }

  // Getters
  // -------
  
  public double getLatitude() {
    return latitude;
  }

  public double getLongitude() {
    return longitude;
  }

}

File: src/main/scala/tweetzor2/core/Location.scala
package tweetzor2.core

import twitter4j.GeoLocation

// Class representing the GPS location of a tweet.
// See `Tweet.scala` for information about the `case class` keywords.
case class Location(
  val latitude: Double,
  val longitude: Double
)

// Companion object and factory method for the `Location` class. 
// See `Tweet.scala` for information about the `object` keyword.
object Location {
  def apply(loc: GeoLocation): Location = {
    Location(
      latitude = loc.getLatitude,
      longitude = loc.getLongitude
    )
  }
}